### DISCUSSION TITLE
Create a default value / oneOf  from items in an array

### DISCUSSION THREAD

**Discussion by TBRSTM** [json-schema-org/community#239] [Q&A]
I have created an array of object, this allow end-user to add several value. 
And I need to have another field called 'default value'. This field would allow me to know whom value from previous array is the default one.
Meaning this would be a list 'supplied' by previous input (dynamic because I a priori don't know which value end-user will input) .
Didn't found any post related, it seems it wouldn't be natively do-able ? 

See example below, I'm trying to point to content of array but it doesn't work.

```
{
    "type": "object",
    "title": "TestObj",
    "properties": {
        "arrayKeyValue": {
            "$ref": "#/definitions/ArrayKeyValue"
        },
        "defaultValue": {
            "$ref": "#/definitions/ArrayKeyValue/items",
            "title": "Default Value"
        }
    },
    "definitions": {
        "ArrayKeyValue": {
            "type": "array",
            "title": "Key/Value",
            "items": {
                "type": "object",
                "properties": {
                    "key": {
                        "type": "string"
                    },
                    "value": {
                        "type": "string"
                    }
                }
            }
        }
    }
}
```

**Comment by GabenGar** [comment]
Do not forget JSON schema only describes the structure of immutable data which are JSON strings. Therefore any dynamic behaviour is not suitable for it. Your logic can only be expressed as a schema when the schemas for array objects are constant values and thus the default value can be expressed as `"anyOf"` keyword. And even then it would validate for any value in that keyword, not specifically the one present in the previous array.
